File: Domain/Entities/EventCategory.cs
using Domain.Common;

namespace Domain.Entities
{
    /// <summary>
    /// Entity that represents the categories available for an event.
    /// </summary>
    public class EventCategory : BaseAuditableEntity
    {
        public Guid CreatorId { get; set; }
        public string? Title { get; set; }
        public string? Description { get; set; }
        public string? Color { get; set; }
        
        public virtual User? Creator { get; set; }

        public virtual ICollection<Event> CategorizedEvents { get; set; } = new HashSet<Event>();
    }
}

File: client-app/src/app/models/eventCategory.ts

export interface EventCategory {
    id: string;
    title: string;
    description: string
    color: string;
    creatorId: string;
}

export const emptyEventCategory = (): EventCategory => ({
    id: '',
    title: '',
    description: '',
    color: '',
    creatorId: '',
});